Python/pylifecycle.c

```diff
@@ -639,6 +643,13 @@ pycore_create_interpreter(_PyRuntimeState *runtime,
         return status;
     }
 
+    // initialize the interp->obmalloc state.  This must be done after
+    // the settings are loaded (so that feature_flags are set) but before
+    // any calls are made to obmalloc functions.
+    if (!_PyMem_init_obmalloc(interp, runtime)) {
+        return  _PyStatus_NO_MEMORY();
+    }
```

Comment:
If we statically initialized `obmalloc_state_main` then we wouldn't need to do this, would we?